src/plugins/ui_feature/ui_sheet.ts

```diff
@@ -170,4 +171,53 @@ export class SheetUIPlugin extends UIPlugin {
     }
     return CommandResult.Success;
   }
+
+  private autoResizeRows(sheetId: UID, rows: HeaderIndex[]) {
+    const rowSizes: (number | null)[] = [];
+    for (const row of rows) {
+      let evaluatedRowSize = 0;
+      for (const id of this.getters.getRowCells(sheetId, row)) {
+        const cell = this.getters.getCellById(id);
+        if (!cell) {
+          continue;
+        }
+        const position = this.getters.getCellPosition(cell.id);
+        const colSize = this.getters.getColSize(sheetId, position.col);
+
+        if (cell.isFormula) {
+          const content = this.getters.getEvaluatedCell(position).formattedValue;
+          const evaluatedSize = getCellContentHeight(this.ctx, content, cell?.style, colSize);
+          if (evaluatedSize > evaluatedRowSize && evaluatedSize > DEFAULT_CELL_HEIGHT) {
+            evaluatedRowSize = evaluatedSize;
+          }
+        } else {
+          const content = cell.content;
+          const dynamicRowSize = getCellContentHeight(this.ctx, content, cell?.style, colSize);
+          // Only keep the size of evaluated cells if it's bigger than the dynamic row size
+          if (dynamicRowSize >= evaluatedRowSize && dynamicRowSize > DEFAULT_CELL_HEIGHT) {
+            evaluatedRowSize = 0;
+          }
+        }
+      }
+      rowSizes.push(evaluatedRowSize || null);
+    }
+
+    if (rowSizes.some(isNotNull)) {
+      for (let i = 0; i < rows.length; i++) {
+        this.dispatch("RESIZE_COLUMNS_ROWS", {
+          elements: [rows[i]],
+          dimension: "ROW",
+          size: rowSizes[i],
+          sheetId,
+        });
+      }
+    } else {
+      this.dispatch("RESIZE_COLUMNS_ROWS", {
+        elements: rows,
+        dimension: "ROW",
+        size: null,
+        sheetId,
+      });
+    }
```

Comment:
you could also group rows by size to avoid dispatching once for every row in case there's an evaluated bazar to resize.

Something like
```ts
const groupedSizes = new Map<number | null, HeaderIndex[]>(rowSizes.map((size) => [size, []]));
for (let i = 0; i < rowSizes.length; i++) {
  groupedSizes.get(rowSizes[i])?.push(rows[i]);
}
for (const [size, rows] of groupedSizes) {
  this.dispatch("RESIZE_COLUMNS_ROWS", {
    elements: rows,
    dimension: "ROW",
    size,
    sheetId,
  });
}
```